cdk/lib/mobile-save-for-later.ts

```diff
@@ -1,19 +1,134 @@
 import { join } from "path";
+import { GuApiGatewayWithLambdaByPath } from "@guardian/cdk";
 import type { GuStackProps } from "@guardian/cdk/lib/constructs/core";
 import { GuStack } from "@guardian/cdk/lib/constructs/core";
+import { GuLambdaFunction } from "@guardian/cdk/lib/constructs/lambda";
 import type { App } from "aws-cdk-lib";
+import { Duration } from "aws-cdk-lib";
+import { PolicyStatement } from "aws-cdk-lib/aws-iam";
+import { Runtime } from "aws-cdk-lib/aws-lambda";
 import { CfnInclude } from "aws-cdk-lib/cloudformation-include";
 
+export interface MobileSaveForLaterProps extends GuStackProps {
+  certificateId: string;
+  domainName: string;
+  hostedZoneName: string;
+  hostedZoneId: string;
+  identityApiHost: string;
+}
+
 export class MobileSaveForLater extends GuStack {
-  constructor(scope: App, id: string, props: GuStackProps) {
+  constructor(scope: App, id: string, props: MobileSaveForLaterProps) {
     super(scope, id, props);
+
     const yamlTemplateFilePath = join(
       __dirname,
       "../..",
       "mobile-save-for-later/conf/cfn.yaml"
     );
-    new CfnInclude(this, "YamlTemplate", {
+    const yamlDefinedResources = new CfnInclude(this, "YamlTemplate", {
       templateFile: yamlTemplateFilePath,
     });
+
+    const app = "mobile-save-for-later";
+
+    const commonLambdaProps = {
+      runtime: Runtime.JAVA_8, // We should upgrade to Java 11 in a future PR
+      app,
+      fileName: `${app}.jar`,
+      // N.B. these lambdas previously used very specific, low memory sizes (384 & 394); odd choices for a Scala Lambda
+    };
+
+    const commonEnvironmentVariables = {
+      App: app,
+      Stack: this.stack,
+      Stage: this.stage,
+      IdentityApiHost: props.identityApiHost,
+    };
+
+    const saveArticlesLambda = new GuLambdaFunction(
+      this,
+      "save-articles-lambda",
+      {
```

Comment:
These properties should align with: 

https://github.com/guardian/mobile-save-for-later/blob/09912344a7cba207687f3cda03ed7338b39396e0/mobile-save-for-later/conf/cfn.yaml#L189-L214